core/server/common/src/main/java/alluxio/master/journal/ufs/UfsJournalLogWriter.java

```diff
@@ -395,12 +394,18 @@ public synchronized void flush() throws IOException, JournalClosedException {
 
   @Override
   public synchronized void close() throws IOException {
-    Closer closer = Closer.create();
+    if (mClosed) {
+      return;
+    }
+    mGarbageCollector.close();
     if (mJournalOutputStream != null) {
-      closer.register(mJournalOutputStream);
+      try {
+        mJournalOutputStream.close();
```

Comment:
Out of curiosity, would it be possible to move to a try-with-resource implementation when using the `mJournalOutputStream`? i.e. a new output stream would be created and closed every time it is needed. Or would it be too expensive? Or perhaps it is necessary that the output stream live across multiple method calls?
I find try-with-resource to be more legible and safer as client/stream lifetime is automatically managed.

Reply:
yeah it'd be prohibitively slow, since every stream creation means a call to UFS.
Also, not every UFS supports appends.